src/main/java/org/opensearch/security/privileges/UserAttributes.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+package org.opensearch.security.privileges;
+
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Iterables;
+
+import org.opensearch.security.user.User;
+
+/**
+ * This code was moved over from ConfigModelV7.
```

Comment:
I like this refactor, might be good to pull this change into a separate PR to lower the surface area in this change.  I imagine it would be merged relatively easily

Reply:
Ok. Do you think we should file two separate PRs for this and for change in ProtectedIndexAccessEvaluator suggested below? I am fine with separate PRs, but these will be really tiny - and likely we will get a couple of more of such changes.

Reply:
+1 to a separate PR. IMO it helps with reviewing this change to isolate the changes most relevant to optimized privileges evaluation and have a separate PR for refactoring these methods into a UserAttributes class.

Reply:
PR in https://github.com/opensearch-project/security/pull/4416